README.md

```diff
@@ -1,64 +1,375 @@
+
+    ```Java
+
+        private ConfidentialClientApplication createClientApplication() throws MalformedURLException {
+            return ConfidentialClientApplication.builder(clientId, ClientCredentialFactory.create(clientSecret))
+                                                .authority(authority)
+                                                .build();
+        }...
+
+          SilentParameters parameters = SilentParameters.builder(
+                        Collections.singleton(scope),
+                        result.account()).build();
+
+                CompletableFuture<IAuthenticationResult> future = app.acquireTokenSilently(parameters);
+                ...
+
+        storeTokenCacheInSession(httpRequest, app.tokenCache().serialize());
+        ...
+    ```
+
+    Important things to notice:
+
+    - We create a `ConfidentialClientApplication` using **MSAL Build Pattern** passing the `clientId`, `clientSecret` and `authority` in the builder. This `ConfidentialClientApplication` will be responsible of acquiring access tokens later in the code.
+    - `ConfidentialClientApplication` also has a token cache, that will cache [access tokens](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens) and [refresh tokens](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow#refresh-the-access-token) for the signed-in user. This is done so that the application can fetch access tokens after they have expired without prompting the user to sign-in again.
+
+3. **AuthFilter** class
+
+    Contains methods for session and state management.
+
+### msal-obo-sample
+
+1. **ApiController** class
+
+    Contains the api(graphMeApi) to trigger the obo flow. The graphMeApi method gets the obo access token using **MsalAuthHelper**. The `callMicrosoftGraphEndPoint` method calls the Microsoft graph API using obo token.
+
+    ```Java
+    String oboAccessToken = msalAuthHelper.getOboToken("https://graph.microsoft.com/.default");
+
+        return callMicrosoftGraphMeEndpoint(oboAccessToken);
+    ```
+
+    Important things to notice:
+
+    - The **scope** [.default](https://docs.microsoft.com/en-us/azure/active-directory/developv2-permissions-and-consent#the-default-scope) is a built-in scope for every application that refers to the static list of permissions configured on the application registration. In our scenario here, it enables the user to grant consent for permissions for both the Web API and the downstream API (Microsoft Graph). For example, the permissions for the Web API and the downstream API (Microsoft Graph) are listed below:
+             - Web Api sample (access_as_user)
+             - Microsoft Graph (user.read)
+
+    - When you use the `.default` scope, the end user is prompted for a combined set of permissions that include scopes from both the **Web Api** and **Microsoft Graph**.
+
+2. **SecurityResourceServerConfig** class
+
```

Comment:
Can we mention one line here that Spring Security is being used for token validation.